managedtenants/bundles/bundle_builder.py

```diff
@@ -41,233 +31,34 @@ def __init__(
             "managedtenants-bundle-builder",
             level=logging.DEBUG if debug else logging.INFO,
         )
-        self._validate_dir_structure()
 
-    def build_push_bundle_images_with_deps(
-        self,
-        versions,
-        hash_string,
-    ):
+    def build_and_push_all(self, bundles, hash_string):
         """
-        Builds and pushes bundle images in the given addon directory.
-
-        :param versions: A list versions to consider
-        :param hash_string: A string to be used in the created image tag.
-
-        :return: A list of pushed bundle images
+        Builds all the bundles. Also sets the bundle.image field.
         """
-        all_inner_addons = get_subdirs(path=self.addon_dir)
-        addon_images = []
-        for inner_addon in all_inner_addons:
-            addon_images.extend(
-                self._build_push_bundle_images(
-                    addon=inner_addon,
-                    versions=versions,
-                    hash_string=hash_string,
-                )
+        bundles = self._build_all(bundles, hash_string)
+        return self._push_all(bundles)
+
+    def _build_all(self, bundles, hash_string):
+        for bundle in bundles:
+            image = Image(
+                f"{self.docker_api.registry}/"
+                f"{bundle.bundle_repo_name()}:{bundle.version}-{hash_string}"
             )
-        return addon_images
-
-    def _build_push_bundle_images(
-        self,
-        addon,
-        hash_string,
-        versions=None,
-    ):
-        """
-        :param addon: A Posix directory path
-        :param versions: A list versions to consider
-        :param hash_string: A string to be used in the created image tag.
+            bundle.set_image(image)
 
-        :return: A list of pushed bundle images
-        """
-        images = []
-        main_addon = self._is_main_addon(addon)
-        # The main addon lives inside the "main" directory, so
-        # we look at its parents name to get the addon name.
-        # (sblaisdo) MTSRE-225: prepend deps /w addon_id to avoid quay
-        # collisions
-        addon_id = addon.parent.name
-        addon_name = addon_id if main_addon else f"{addon_id}-{addon.name}"
+            self.log.info(f"Building {bundle}.")
+            _ = self.docker_api.build_bundle(bundle)
```

Comment:
Why isn't there a `try..catch` here? If the intention is to just fail anyway then I don't see why `bundles` should be returned for pipelining.

Reply:
added try/catch, not returning bundles it is indeed not very useful.